kafka-connect/kafka-connect/src/main/java/org/apache/iceberg/connect/channel/CommitterImpl.java

```diff
@@ -74,12 +74,15 @@ public int compare(TopicPartition o1, TopicPartition o2) {
     }
   }
 
-  private boolean hasLeaderPartition(Collection<TopicPartition> currentAssignedPartitions) {
+  private boolean hasLeaderPartition(
```

Comment:
Maybe change the parameter name to `onlyIfStable` to be a little bit more clear? Alternatively the method could return an enum indicating the state.